src/bin/cloudmqtt-test-client.rs

```diff
@@ -160,14 +176,20 @@ async fn main() {
         }
     }
 
+    let timeout = std::time::Duration::from_millis(50);
     loop {
-        let _packet = match packet_stream.next().await {
-            Some(Ok(packet)) => packet,
-            None => {
-                eprintln!("Stream ended, stopping");
-                break;
-            }
-            Some(Err(error)) => print_error_and_quit(format!("Stream errored: {error}")),
-        };
+        let next_packet_fut = tokio::time::timeout(timeout, packet_stream.next());
```

Comment:
This timeout makes some tests succeed, but that's most likely not what we want.
Unfortunately, it seems that stdin does not get closed (or it doesn't result in what we expect here, which is the match below resulting in `None` and the loop to be broken then.

I fail to debug what causes this behaviour. Stdin is clearly dropped in the parent process, which _should_ result in it being closed and this loop to be exited. But it does not happen and I don't understand why.